pkg/rancher-desktop/config/settingsImpl.ts

```diff
@@ -447,32 +468,38 @@ export const updateTable: Record<number, (settings: any) => void> = {
   },
 };
 
-function migrateSettingsToCurrentVersion(settings: Settings) {
+function migrateSettingsToCurrentVersion(settings: Record<string, any>): Settings {
   if (Object.keys(settings).length === 0) {
     return defaultSettings;
   }
-  let loadedVersion = settings.version || 0;
+  settings = migrateSpecifiedSettingsToCurrentVersion(settings) as Settings;
 
-  if (loadedVersion < CURRENT_SETTINGS_VERSION) {
-    for (; loadedVersion < CURRENT_SETTINGS_VERSION; loadedVersion++) {
-      if (updateTable[loadedVersion]) {
-        updateTable[loadedVersion](settings);
-      }
+  return _.defaultsDeep(settings, defaultSettings);
+}
+
+export function migrateSpecifiedSettingsToCurrentVersion(settings: Record<string, any>): RecursivePartial<Settings> {
```

Comment:
Please add a doc comment on an exported function.

I guess this exists because we expect the given settings to be partial, whereas `migrateSettingsToCurrentVersion` is always the complete settings?